Kickstarter-iOS/DataSources/PledgeDataSource.swift

```diff
@@ -12,34 +12,36 @@ final class PledgeDataSource: ValueCellDataSource {
 
   // MARK: - Load
 
-  func load(project: Project, reward: Reward, isLoggedIn: Bool) {
+  func load(data: PledgeViewData) {
+    self.clearValues()
+
     self.appendRow(
-      value: reward,
+      value: data.reward,
       cellClass: PledgeDescriptionCell.self,
       toSection: Section.project.rawValue
     )
 
     self.appendRow(
-      value: (project, reward),
+      value: (data.project, data.reward),
       cellClass: PledgeAmountCell.self,
       toSection: Section.inputs.rawValue
     )
 
-    if reward.shipping.enabled {
+    if data.reward.shipping.enabled {
```

Comment:
I'm wondering if we should move this logic to the view model? 🤔 I know we have a test for this in the data source tests but it seems to me like something that we should keep in the view model in case we decide to swap out this table view for a stack view, let's say.

Reply:
Do you mean appending the row logic Isabel? Can you elaborate a bit more on this?

I think the way this is currently defined here should be correct since this is the whole purpose of the data source (to define how many rows a table view or collection view or any list UI will show). 

If we decide to switch architectures then this logic would probably move on to the controller's `viewDidLoad` where we would construct the stack view (unless we do some conditional UI loading there).

But I might not follow your question just right 🤔

Reply:
Oh I just mean instead of doing `data.reward.shipping.enabled` here, keeping that in the view model and just passing the actual value `isShippingEnabled` via the `PledgeViewData`:

```
if data.isShippingEnabled {
// append shipping row
}
```

That way we're separating out **how** we determine whether the cell is added or not (ie. the business logic) from the actual hiding/showing of the cell.

Does that make sense?

Reply:
Oh right, so you're suggesting we pass only the information that's needed right? I think the only downside to that would be duplication of data seeing as we have to pass `reward` to the datasource anyway?